src/palace/manager/api/admin/controller/report.py

```diff
@@ -3,23 +3,80 @@
 
 import flask
 from flask import Response
+from sqlalchemy import not_, select
 from sqlalchemy.orm import Session
 
+from palace.manager.api.admin.model.inventory_report import (
+    InventoryReportCollectionInfo,
+    InventoryReportInfo,
+)
+from palace.manager.api.admin.problem_details import ADMIN_NOT_AUTHORIZED
 from palace.manager.celery.tasks.generate_inventory_and_hold_reports import (
     generate_inventory_and_hold_reports,
 )
 from palace.manager.core.problem_details import INTERNAL_SERVER_ERROR
+from palace.manager.integration.goals import Goals
 from palace.manager.sqlalchemy.constants import MediaTypes
 from palace.manager.sqlalchemy.model.admin import Admin
+from palace.manager.sqlalchemy.model.collection import Collection
+from palace.manager.sqlalchemy.model.integration import (
+    IntegrationConfiguration,
+    IntegrationLibraryConfiguration,
+)
 from palace.manager.sqlalchemy.model.library import Library
 from palace.manager.util.log import LoggerMixin
-from palace.manager.util.problem_detail import ProblemDetail
+from palace.manager.util.problem_detail import ProblemDetail, ProblemDetailException
 
 
 class ReportController(LoggerMixin):
     def __init__(self, db: Session):
         self._db = db
 
+    def inventory_report_info(self) -> Response:
+        """InventoryReportInfo response of reportable collections for a library.
+
+        returns: Inventory report info response, if the library exists and
+            the admin is authorized.
+            Otherwise, return a 404 response if the library does not exist
+            or raise an ADMIN_NOT_AUTHORIZED ProblemDetailException, if the
+            admin is not authorized.
+        """
+        library: Library | None = getattr(flask.request, "library")
+        if library is None:
+            return Response(status=404)
+
+        admin: Admin = getattr(flask.request, "admin")
+        if not admin.is_librarian(library):
+            raise ProblemDetailException(ADMIN_NOT_AUTHORIZED)
+
+        collections = self._db.scalars(
+            select(Collection)
+            .join(IntegrationConfiguration)
+            .join(IntegrationLibraryConfiguration)
+            .where(
+                IntegrationLibraryConfiguration.library_id == library.id,
+                IntegrationConfiguration.goal == Goals.LICENSE_GOAL,
+                not_(
+                    IntegrationConfiguration.settings_dict.contains(
+                        {"include_in_inventory_report": False}
```

Comment:
I got most of this `select` from the code for generating the reports, but I'm not sure where the "include_in_inventory_report" setting is coming from. I do see it [here in OPDS collection settings](https://github.com/ThePalaceProject/circulation/blob/b1ab61ac9a9506d4ccad11efaeb4df5e3998a84b/src/palace/manager/core/opds_import.py#L131-L141), but I don't see it getting set to false on other collection types that probably shouldn't be included in inventory reports. I looked around a bit, but didn't see it set to either `true` or `false`.  Here's the query I used:

```sql
SELECT *
FROM integration_configurations ic
WHERE ic.goal = 'LICENSE_GOAL'
  AND (
    ic.settings @> '{"include_in_inventory_report": true}' OR  
    ic.settings @> '{"include_in_inventory_report": false}'
  )
```

Reply:
So getting this info in a pure SQL query is hard, mostly due to our use of inheritance to set the `include_in_inventory_report` setting. 

Some background on how this is set: 

You are correct this setting is getting set in the OPDS collection settings:
https://github.com/ThePalaceProject/circulation/blob/b1ab61ac9a9506d4ccad11efaeb4df5e3998a84b/src/palace/manager/core/opds_import.py#L131-L141

We don't store the default values for our serialized settings in the database. This is done in the `dict()` method for the Pydantic settings class:
https://github.com/ThePalaceProject/circulation/blob/530f25be8e64f4ef6a62294bcf2b25be8edd225b/src/palace/manager/integration/settings.py#L337-L341

I thought this was better called out in the comments, but I can't find it right now, so we should probably add a better comment adding these details. But we're not storing the defaults when we serialize to the database, so it is easier to update the defaults in the future.

When we are interacting with the settings via the Pydantic class (how I was envisioning we would mostly interact with them, rather then querying, this is transparent, since the defaults get re-filled when the values are loaded from the database).

So the only value that ever gets saved out to the DB is `include_in_inventory_report == False`. So that is why the code that finds the integrations when generating the reports is structured the way it is. 

First it finds any license integrations where `include_in_inventory_report == False` is not in the settings via a query. 
https://github.com/ThePalaceProject/circulation/blob/530f25be8e64f4ef6a62294bcf2b25be8edd225b/src/palace/manager/celery/tasks/generate_inventory_and_hold_reports.py#L66-L79

Then it filters down this list, to only those integrations that actually contain this setting:
https://github.com/ThePalaceProject/circulation/blob/530f25be8e64f4ef6a62294bcf2b25be8edd225b/src/palace/manager/celery/tasks/generate_inventory_and_hold_reports.py#L80-L86

Reply:
Since this code is missing that last filtering step, it will return collections without the setting, which I don't think is what we want here. 

Perhaps since this query is so complicated, it should get broken out to a utility function that both the report generation and the controller can call.